### PR TITLE
Fixes various things related to Kapu Limbs

### PR DESCRIPTION
**Pull request by dwasint**
# About The Pull Request
closes https://github.com/Monkestation/MonkeStation/issues/769
Fixes Lizard mut colors not updating which fixes: char, baja blast, devils, and any other form of skin updating

Fixes damage overlays they now show up again

Fixes new mobs spawning bald

resprites base lizards Tau Ceti spirtes

snouts now layer correctly

adds secondary color to lizards using body marks so we can expand it in the future

adds a forced update flag to regenerate icons which forcibly rebuilds sprites even if they don't need it

fixes lizards being beheaded

sprites are now in a sane pathing for lizard parts

belly color has been moved to its own pref/gene as it functioned really wonky, and allows for more customization

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

## Changelog

:cl:
fix:Lizard mut colors not updating which fixes: char, baja blast, devils, and any other form of skin updating
fix:damage overlays they now show up again
fix:new mobs spawning bald
adds: markings have been split to its own pref/gene for coloring
/:cl:

### CODE REVIEW

File: code/__HELPERS/cmp.dm [hunk 1]
```diff
@@ -136,3 +136,8 @@ GLOBAL_VAR_INIT(cmp_field, "name")
 
 /proc/cmp_mob_realname_dsc(mob/A,mob/B)
 	return sorttext(A.real_name,B.real_name)
+
+/// Orders bodyparts by their body_part value, ascending.
+/proc/cmp_bodypart_by_body_part_asc(obj/item/bodypart/limb_one, obj/item/bodypart/limb_two)
+	return limb_one.body_part - limb_two.body_part
```
Comment (nednaZ) [inline code/__HELPERS/cmp.dm:142]: Could this be done with a define instead?
Seems very specific for a simple subtraction operation.

File: code/modules/mob/living/carbon/human/species.dm [hunk 2]
```diff
@@ -599,326 +693,361 @@ GLOBAL_LIST_EMPTY(roundstart_races)
...
-
-	H.remove_overlay(BODY_BEHIND_LAYER)
-	H.remove_overlay(BODY_ADJ_LAYER)
-	H.remove_overlay(BODY_FRONT_LAYER)
-
-	if(!mutant_bodyparts)
-		return
-
-	var/obj/item/bodypart/head/HD = H.get_bodypart(BODY_ZONE_HEAD)
+// this proc is cancer to look at but basically it reads through all the body parts in mutant_bodyparts and checks visibility if it can't see it removes from the list
```
Comment (nednaZ) [inline code/modules/mob/living/carbon/human/species.dm:784]: Change out the comment for a different one.
Can't have text like that going into the codebase.

**Review by nednaZ — CHANGES_REQUESTED** [review]
Fix your changelog.
"fix", not "fixes"
The belly color changes are not documented in the changelog and are out of scope of a fix PR.

Not seeing major issues.

File: code/__HELPERS/cmp.dm (hunk 1 above)
Comment (dwasint) [reply code/__HELPERS/cmp.dm:142]: yes/no, this gives more freedom for limbs and allows for things like quad arms to render properly if we ever do so. Basically fits with what klimbs was intended for.

**Comment by KoboldCommando** [comment]
There appears to be a red spot that shows up on the neck on the side and rear view. 
image

On female lizards, the jumpsuits show two pixels of your torso. 
image

Personally I'd be cool with making female lizards just wear all the male clothing.  At least until/unless someone wants to implement specific female sprites down the line.

**Comment by nednaZ** [comment]
Was the issue with the red spot/extra pixels fixed?
If so, it's ready for merge

**Comment by dwasint** [comment]
> Was the issue with the red spot/extra pixels fixed? If so, it's ready for merge

yes it was fixed that as it was reported

**Comment by nednaZ** [comment]
No unsolved issues found during testmerges.

**Pull request merged**